velox/common/memory/MappedMemory.cpp

```diff
@@ -338,6 +338,64 @@ std::shared_ptr<MappedMemory> MappedMemory::addChild(
   return std::make_shared<ScopedMappedMemory>(this, tracker);
 }
 
+namespace {
+// Returns the size class size that corresponds to 'size'.
+int32_t sizeClassSize(
+    int32_t size,
+    const std::vector<MachinePageCount>& sizes) {
+  VELOX_CHECK_LE(size, sizes.back());
+  return *std::lower_bound(sizes.begin(), sizes.end(), size);
+}
+} // namespace
+
+void* FOLLY_NULLABLE
+MappedMemory::allocateBytes(uint64_t size, int32_t maxMallocSize) {
+  if (size <= maxMallocSize) {
+    return reinterpret_cast<char*>(::malloc(size));
+  } else if (size <= sizeClassSizes_.back() * kPageSize) {
+    Allocation allocation(this);
+    auto numPages = sizeClassSize(
+        bits::roundUp(size, kPageSize) / kPageSize, sizeClassSizes_);
+    if (allocate(numPages, kMallocOwner, allocation, nullptr, numPages)) {
+      auto run = allocation.runAt(0);
+      VELOX_CHECK_EQ(
+          1,
+          allocation.numRuns(),
+          "A size class alocateBytes must produce one run");
+      allocation.clear();
+      return run.data<char>();
+    }
+    return nullptr;
+  } else {
+    ContiguousAllocation allocation;
+    if (allocateContiguous(
+            bits::roundUp(size, kPageSize) / kPageSize, nullptr, allocation)) {
+      char* data = allocation.data<char>();
+      allocation.reset(nullptr, nullptr, 0);
+      return data;
+    }
+  }
+}
+
+void MappedMemory::freeBytes(
```

Comment:
There is a strict requirement for the caller to allocate and free exact same amount of bytes, otherwise we will have memory leak. This is not easily resolvable though but we should keep this in mind. I wonder if we should have a counter to store currently allocated bytes through this interface to track.